### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by mkoncek**
Resolves: #97

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -107,36 +133,36 @@ exclude-tests-matching:
...
-Project is built with Maven. Simply run:
-[source, shell]
-----
-$ mvn install
-----
-
 == Custom validators
-A custom validator must inherit from the `org.fedoraproject.javapackages.validator.Validator` class and have its name listed in the service file on the class path of the program.
-Custom validators can override functions:
+A custom validator must implement the `org.fedoraproject.javapackages.validator.spi.Validator` interface and have its name listed in the service file on the class path of the program.
```
Comment (mizdebsk) [inline README.adoc:137]: Validators are not listed in service files. Validator factories are.

File: README.adoc [hunk 2]
```diff
@@ -159,3 +185,33 @@ At least one check failed.
 `error`::
 An error occured, for example invalid input or an unexpected state.
 [horizontal!]
+
+=== Invoking custom validators
+If the user wants to run the validator on a list of separate `.java` or `class` files, they need to be passed via the `--cp` or the `--sp` flags.
```
Comment (mizdebsk) [inline README.adoc:190]: In sentence "running validator on something", "something" is the thing validator inspects.

File: README.adoc [hunk 3]
```diff
@@ -66,12 +84,20 @@ Recompilation is caused by any of these conditions:
 
 Recompilation causes the class path directory to be cleaned before the newly compiled classes are placed there.
 
-=== Service file
+=== https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/ServiceLoader.html#deploying-service-providers-on-the-class-path-heading[Service file]
```
Comment (mizdebsk) [inline README.adoc:87]: I would remove the link from section header and add it in the text below.

**Review by mizdebsk — CHANGES_REQUESTED** [review]
`-u` is still used (issue #85), should it be removed as well?
Otherwise it looks good to me except a few minor issues.

**Comment by mkoncek** [comment]
I was wondering if I implement or remove the `-u` feature, but I will postpone it for now.

**Review by mizdebsk — APPROVED** [review]
LGTM

**Pull request merged**